### PR TITLE
Custom Error Message (resolves issue #26)

### PR DESCRIPTION
**Pull request by saikatharryc**
#### Changes : Custom Error Message
To send custom error message for all, 
You can pass `errorMessage` during the plugin registration:
```
fastify.register(require('fastify-rate-limit'), {
  global : false, // default true
  max: 3, // default 1000
  timeWindow: 5000, // default 1000 * 60
  cache: 10000, // default 5000
  whitelist: ['127.0.0.1'], // default []
  redis: new Redis({ host: '127.0.0.1' }), // default null
  skipOnError: true, // default false
  keyGenerator: function(req) { /* ... */ }, // default (req) => req.raw.ip
  errorMessage: 'This is your custom error message' //default 'Rate limit exceeded, retry in 1 minute'
})
```

for each route, you can override custom message for each, by supplying `errorMessage` like below:
```
fastify.get('/custom', {
  config: {
    rateLimit: {
      max: 3,
      timeWindow: '1 minute',
      errorMessage: 'Please try after sometime.'
    }
  }
},(req, reply) => {
    reply.send({ hello: 'Hello World!' })
})
```

### CODE REVIEW

**Comment by saikatharryc** [comment]
#26

File: README.md [hunk 1]
```diff
@@ -68,6 +69,7 @@ fastify.register(require('fastify-rate-limit'), {
 You can pass a Redis client here and magically the issue is solved. To achieve the maximum speed, this plugins requires the use of [`ioredis`](https://github.com/luin/ioredis)
 - `skipOnError`: if `true` it will skip errors generated by the storage (eg, redis not reachable).
 - `keyGenerator`: a function to generate a unique identifier for each incoming request. Defaults to `(req) => req.ip`, the IP is resolved by fastify using `req.connection.remoteAddress` or `req.headers['x-forwarded-for']` if [trustProxy](https://www.fastify.io/docs/master/Server/#trustproxy) option is enabled. Use it if you want to override this behavior
+- `errorMessage` : You can pass a string, as a custom error message which will override the default error message.
```
Comment (mcollina) [inline README.md]: I would write the default here

File: index.js [hunk 2]
```diff
@@ -127,16 +148,23 @@ function buildRouteRate (pluginComponent, params, routeOptions) {
...
+        res
+          .code(429)
           .header('X-RateLimit-Limit', params.max)
           .header('X-RateLimit-Remaining', 0)
           .header('Retry-After', params.timeWindow)
           .send({
             statusCode: 429,
             error: 'Too Many Requests',
-            message: `Rate limit exceeded, retry in ${after}`
+            message
```
Comment (mcollina) [inline index.js]: Have you changed the linting in this file? Would you mind removing the linting changes, and limit this to the actual change?

File: index.d.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
-import * as http from 'http';
-import * as fastify from 'fastify';
+import * as http from "http";
+import * as fastify from "fastify";
```
Comment (mcollina) [inline index.d.ts]: Can you please restore the previous linting?

**Comment by pc-jedi** [comment]
Am I right, that this PR is just making the error message customizable?
#26 Was about having different JSON scheme then:
```json
{
  "error": "Too Many Requests",
  "message": "Rate limit exceeded, retry in 1 minute",
  "statusCode": 429
}
```

For example:
```json
{
  "error": "Too Many Requests",
  "message": "Rate limit exceeded, retry in 1 minute",
  "code": 429,
  "request": "/api/v1/foobar/",
  "time": "Sat Aug 03 2019 10:42:57 GMT+0200 (Central European Summer Time)"
}
```

**Comment by saikatharryc** [comment]
@pc-jedi  yes you are right.

**Pull request closed without merging**